client/checkout/utils/upe.js

```diff
@@ -300,3 +300,71 @@ export const getStripeElementOptions = (
 
 	return options;
 };
+
+/**
+ * Converts form fields object into Stripe `shipping` object.
+ *
+ * @param {Object} fields Object mapping checkout shipping fields to values.
+ * @return {Object} Stripe formatted `shipping` object.
+ */
+export const getShippingDetails = ( fields ) => {
+	// Shipping address is needed by Afterpay. If available, use shipping address, else fallback to billing address.
+	if (
+		fields.ship_to_different_address &&
+		'1' === fields.ship_to_different_address
+	) {
+		return {
+			name:
+				`${ fields.shipping_first_name } ${ fields.shipping_last_name }`.trim() ||
+				'-',
+			address: {
+				country: fields.shipping_country || '-',
+				line1: fields.shipping_address_1 || '-',
+				line2: fields.shipping_address_2 || '-',
+				city: fields.shipping_city || '-',
+				state: fields.shipping_state || '-',
+				postal_code: fields.shipping_postcode || '-',
+			},
+		};
+	}
+	return {
```

Comment:
That's pretty much a code duplicate. We can refactor this a bit like this:

```javascript
let billingAsShippingAddress = getBillingDetails(fields);
delete billingAsShippingAddress.email;

return billingAsShippingAddress;
```

Reply:
Thanks for this solution! I was feeling a bit itchy about the repetition. I will keep this in mind for future patterns too. 🙏 

Resolved here: 07c54da